tests/js/sentry-test/performance/utils.ts

```diff
@@ -0,0 +1,169 @@
+import {RawSpanType} from 'sentry/components/events/interfaces/spans/types';
+import {EntryType, EventOrGroupType, EventTransaction} from 'sentry/types';
+
+import {PerformanceDetectorData} from './../../../../static/app/types/event';
+
+export enum ProblemSpan {
+  PARENT = 'parent',
+  OFFENDER = 'offender',
+}
+
+export const EXAMPLE_TRANSACTION_TITLE = '/api/0/transaction-test-endpoint/';
+
+type AddSpanOpts = {
+  endTimestamp: number;
+  startTimestamp: number;
+  childOpts?: AddSpanOpts[];
```

Comment:
So the best solution I could think of to allow for nested children is by `childOpts` to allow for recursively building out the event. I can see the calls to `addSpan` getting quite gross if we need to build some deeply nested spans, so I'm open to suggestions on how to make this API cleaner.

Reply:
One idea is that you could make a `SpanBuilder` class. Then it could have an `addChildSpan()` method. Something like:

```ts
const builder = new TransactionEventBuilder();

const span = new SpanBuilder({...})
span.addChildSpan(new SpanBuilder({...})

builder.addSpan(span);
```

But since this is just a test util, don't think it matters too much. I think this looks great!

Reply:
I like this idea! I'll add this in